File: lib/temp/sample_component.js
Netzke.classes.SampleComponent = Ext.extend(Ext.TabPanel, 
  Ext.apply(Netzke.componentMixin(Ext.TabPanel), {
    // ...
    "addTab": function(title){
      this.add({title: title});
    },
    
    // Some event handler
    "onButtonClick": function(){
      var args = {...}
      
      // optionally accepts a callback-function
      this.createTab(args);
    }
}));

File: lib/temp/sample_component.rb
class SampleComponent < Netzke::Base
  # ...
  
  js_method :add_tab, <<-JS
    function(title){
      this.add({title: title});
    }
  JS
  
  # ...
  
  endpoint :create_tab do |params|
    { :add_tab => "New Tab"}
  end
  
end

